### PR TITLE
initial book entry for concolic

### PR DESCRIPTION
**Pull request by julihoh**
TODO:

- [x] release concolic crates to make links work.
- [x] I assumed in the markdown that the next version of libafl will be 0.6.0. if this is not the case, I'll have to update the links.

### CODE REVIEW

**Comment by julihoh** [comment]
This is ready to be merged from my side @andreafioraldi @domenukk

File: docs/src/advanced_features/concolic/concolic.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+In hybrid fuzzing, we combine this tracing + solving approach with more traditional fuzzing techniques.
+
+## Concolic Tracing in LibAFL, SymCC and SymQEMU
+The concolic tracing support in LibAFL is implemented using SymCC.
+SymCC is a compiler plugin for clang that can be used as a drop-in replacement for a normal C or C++ compiler.
+SymCC will instrument the compiled code with callbacks into a runtime that can be supplied by the user.
+These callbacks allow the runtime to construct a trace that similar to the previous example.
+
+### SymCC and it's Runtimes
```
Comment (domenukk) [inline docs/src/advanced_features/concolic/concolic.md:89]: it's vs its :P

File: docs/src/advanced_features/concolic/concolic.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+It should be noted, however, that the 'quality' of the generated expressions can be significantly worse and SymQEMU generally produces significantly more and signficantly more convoluted expressions than SymCC.
+Therefore, it is recommended to use SymCC over SymQEMU when possible.
+
+#### Using SymCC
+The target needs to be instrumented ahead of fuzzing using SymCC.
+How exactly this is done does not matter.
+However, the SymCC compiler needs to be made aware of the location of the runtime that it should instrument against.
+This is done by setting the `SYMCC_RUNTIME_DIR` environment variable to the directory which contains the runtime (typically the `target/(debug|release)` folder of your runtime crate).
+
+The example hybrid fuzzer instruments the target in it's [`build.rs` build script](https://github.com/AFLplusplus/LibAFL/blob/main/fuzzers/libfuzzer_stb_image_concolic/fuzzer/build.rs#L50).
```
Comment (domenukk) [inline docs/src/advanced_features/concolic/concolic.md:135]: it's

File: docs/src/advanced_features/concolic/concolic.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+It does this by cloning and building a copy of SymCC and then using this version to instrument the target.
+The [`symcc_libafl` crate](https://docs.rs/symcc_libafl) contains helper functions for cloning and building SymCC.
+
+Make sure you satisfy the [build requirements](https://github.com/eurecom-s3/symcc#readme) of SymCC before attempting to build it.
+
+#### Using SymQEMU
+Build SymQEMU according to it's [build instructions](https://github.com/eurecom-s3/symqemu#readme).
```
Comment (domenukk) [inline docs/src/advanced_features/concolic/concolic.md:142]: it is?

File: docs/src/advanced_features/concolic/concolic.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+Make sure to let it know the path to your runtime by setting `--symcc-build` argument to the `configure` script to the path of your runtime.
```
Comment (domenukk) [inline docs/src/advanced_features/concolic/concolic.md:143]: Everything starts with "Make sure"

**Review by domenukk — APPROVED** [review]
Sweet intro! :)

**Comment by julihoh** [comment]
fixed all nits + some more. can we get this merged + published before tonight? (tonight's the deadline for GSoC 😬)

**Pull request merged**